src/Bridges/Constraint/bridges/split_hyperrectangle.jl

```diff
@@ -88,6 +88,9 @@ function MOI.get(
     ::MOI.ConstraintFunction,
     bridge::SplitHyperRectangleBridge,
 )
+    # We cache the function to simplify cases in which the box is (-Inf, Inf).
+    # This contributes some memory overhead, so if someone complains in future
```

Comment:
I don't agree that we should wait for someone to complain, it's a TODO. See the warning in https://jump.dev/MathOptInterface.jl/dev/tutorials/bridging_constraint/#Attribute-retrieval that was added in https://github.com/jump-dev/MathOptInterface.jl/pull/1952/

Reply:
Fully agree that in the common case, we shouldn't cache the functions. But this is a pretty niche bridge that in all likelihood won't get used much. I don't want to spend much time optimizing if for an edge-case.

Reply:
You don't have to do it but because it's a convention, we should at least mark it as a TODO or FIXME otherwise we're being inconsistent.
Memory overhead is quite hard to track down so users won't come complaining about that.
It's not that niche though, box constraints are very common in control.
Note that the bookkeeping needed to implement this will be useful for `ConstraintPrimal` and `ConstraintDual` anyway.
Anyway, no big deal, I can do that in a follow up PR to clean up the TODO

Reply:
I've been omitting `Constraint{Primal/Dual}{Start}` for simplicity. We can come back to them. It's a lower priority.

Reply:
`ConstraintFunction` has a much lower priority than `ConstraintDual`. `ConstraintFunction` is mostly used in MOI tests but users are recommended to use a cache if they need to query the model. I feel the issue is that the bridge tests need `ConstraintFunction` to be implemented. Maybe we could add an option for the test to not test the `ConstraintFunction` attribute ? If you just cache the constraint function as in this PR, it's not like this is testing anything anyway. It seems that the fact that `MOI.Bridges.runtests` forces `ContraintFunction` to be implemented might be annoying. You might want to first make it work without having to worry about that attribute

Reply:
`Constraint{Function/Set}` are really useful because they allow printing of the bridged models. If you have a cache, It's harder to see what the bridge is really doing.

Reply:
Okay, I'll add the starts now.

Reply:
What do you mean, it will print the same as the cache no?

Reply:
In the `Bridges.runtests`, I often print the bridge when I'm writing it as a way to check things. But yeah, I guess we could make the test optional, but at this point, every bridge supports it.